### PR TITLE
Add top bar for the chat

### PR DESCRIPTION
**Pull request by jrainville**
Adds the top bar for the chat including hooks to the channel name and new colors for the icon

image

Disclaimer: the tree dots in the top right do nothing for now but have the MouseArea

### CODE REVIEW

**Comment by richard-ramos** [comment]
Nice!

File: src/app/chat/view.nim [hunk 1]
```diff
@@ -75,7 +85,9 @@ QtObject:
       result = self.chats.chats.findById(channel)
     else:
       self.model.join(channel)
-      result = self.chats.addChatItemToList(ChatItem(id: channel, name: channel))
+      randomize()
+      let randomColorIndex = rand(channelColors.len - 1)
```
Comment (richard-ramos) [inline src/app/chat/view.nim:89]: I think the chat colors are deterministic. I saw the color being part of the message signal

File: src/app/chat/views/channels_list.nim [hunk 2]
```diff
@@ -66,6 +86,12 @@ QtObject:
     else:
       result = idx
 
+  proc getChannelByName*(self: ChannelsList, name: string): ChatItem =
+    for chat in self.chats:
+      if chat.name == name:
+        return chat
+    raise newException(OSError, fmt"No chat found with the name {name}")
```
Comment (richard-ramos) [inline src/app/chat/views/channels_list.nim:93]: Is this type of exception captured in the nim side, and logged?

File: src/app/chat/view.nim (hunk 1 above)
Comment (jrainville) [reply src/app/chat/view.nim:89]: They're not
I found the code in status-react. I was actually surprised. 
https://github.com/status-im/status-react/blob/develop/src/status_im/chat/models.cljs#L90

I tried opening #test in mobile and my simulator and had different colors.

File: src/app/chat/views/channels_list.nim (hunk 2 above)
Comment (jrainville) [reply src/app/chat/views/channels_list.nim:93]: Not sure. I catch it on the other side of the call anyway

**Pull request merged**